src/nixos.rs

```diff
@@ -720,6 +722,72 @@ impl OsReplArgs {
     }
 }
 
+impl OsEditArgs {
+    fn run(self) -> Result<()> {
+        let editor = env::var("EDITOR").unwrap_or_else(|_| "nano".to_string());
+
+        // Use NH_OS_FLAKE if available, otherwise use the provided installable
+        let target_installable = if let Ok(os_flake) = env::var("NH_OS_FLAKE") {
+            debug!("Using NH_OS_FLAKE: {}", os_flake);
+
+            let mut elems = os_flake.splitn(2, '#');
+            let reference = match elems.next() {
+                Some(r) => r.to_owned(),
+                None => return Err(eyre!("NH_OS_FLAKE missing reference part")),
+            };
+            let attribute = elems
+                .next()
+                .map(crate::installable::parse_attribute)
+                .unwrap_or_default();
+
+            Installable::Flake {
+                reference,
+                attribute,
+            }
+        } else {
+            self.installable
+        };
+
+        match target_installable {
+            Installable::File { path, .. } => {
+                let flake_path = path.join("flake.nix");
+                let str_path = flake_path
+                    .to_str()
+                    .ok_or_else(|| eyre!("Edit path is not valid UTF-8"))?;
+
+                Command::new(editor)
+                    .arg(str_path)
+                    .with_required_env()
+                    .show_output(true)
+                    .run()?
+            }
+            Installable::Flake { reference, .. } => {
+                Command::new("nix")
+                    .arg("flake")
+                    .arg("prefetch")
+                    .arg("-o")
+                    .arg("result")
+                    .arg(reference)
+                    .with_required_env()
+                    .show_output(true)
+                    .run()?;
+
+                Command::new(editor)
+                    .arg("result/flake.nix")
+                    .with_required_env()
+                    .show_output(true)
+                    .run()?;
+
+                std::fs::remove_file("result")?
```

Comment:
```suggestion
                std::fs::remove_dir_all("result")?;
```

`nix flake prefetch -o result` creates a directory, not a file. Using `remove_file` will always fail and leave junk directories.

Reply:
This felt wrong because I'm trying to delete the symlink and not the actual directory which is read only. The default behavior of remove_file is to only delete the symlink when it is a link.